webservice/src/components/ProfessorFunctionalities.jsx

```diff
@@ -0,0 +1,306 @@
+import AddCircleIcon from '@material-ui/icons/AddCircle';
+import Slider from '@material-ui/core/Slider';
+import withStyles from '@material-ui/core/styles/withStyles';
+import Typography from '@material-ui/core/Typography';
+import DeleteIcon from '@material-ui/icons/Delete';
+import Slide from '@material-ui/core/Slide';
+import Dialog from '@material-ui/core/Dialog';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogActions from '@material-ui/core/DialogActions';
+import MenuItem from '@material-ui/core/MenuItem';
+
+export default function ProfessorFunc(props) {
+  return (
+    <Grid
+      container
+      spacing={0}
+      alignItems="center"
+      justify="center"
+      direction="row"
+      noValidate
+      autoComplete="off"
+    >
+      <NewTemplateSlider
+        funcNewTemplate={props.funcNewTemplate}
+        adminGroups={props.adminGroups}
+      />
+    </Grid>
+  );
+}
+
+const Transition = React.forwardRef(function Transition(props, ref) {
+  return <Slide direction="up" ref={ref} {...props} />;
+});
+
+const iOSBoxShadow =
+  '0 3px 1px rgba(0,0,0,0.1),0 4px 8px rgba(0,0,0,0.13),0 0 0 1px rgba(0,0,0,0.02)';
+
+const ram = [
+  { value: 0.5 },
+  { value: 1 },
+  { value: 1.5 },
+  { value: 2 },
+  { value: 2.5 },
+  { value: 3 },
+  { value: 3.5 },
+  { value: 4 },
+  { value: 4.5 },
+  { value: 5 },
+  { value: 5.5 },
+  { value: 6 },
+  { value: 6.5 },
+  { value: 7 },
+  { value: 7.5 },
+  { value: 8 }
+];
+
+const cpu = [
+  { value: 1 },
+  { value: 2 },
+  { value: 3 },
+  { value: 4 },
+  { value: 5 },
+  { value: 6 },
+  { value: 7 },
+  { value: 8 }
+];
+
+const IOSSlider = withStyles({
+  root: {
+    color: '#3880ff',
+    height: 2,
+    padding: '15px 0'
+  },
+  thumb: {
+    height: 28,
+    width: 28,
+    backgroundColor: '#fff',
+    boxShadow: iOSBoxShadow,
+    marginTop: -14,
+    marginLeft: -14,
+    '&:focus, &:hover, &$active': {
+      boxShadow:
+        '0 3px 1px rgba(0,0,0,0.1),0 4px 8px rgba(0,0,0,0.3),0 0 0 1px rgba(0,0,0,0.02)',
+      // Reset on touch devices, it doesn't add specificity
+      '@media (hover: none)': {
+        boxShadow: iOSBoxShadow
+      }
+    }
+  },
+  active: {},
+  valueLabel: {
+    left: 'calc(-50% + 11px)',
+    top: -22,
+    '& *': {
+      background: 'transparent',
+      color: '#ff0000'
+    }
+  },
+  track: {
+    height: 2
+  },
+  rail: {
+    height: 2,
+    opacity: 0.5,
+    backgroundColor: '#bfbfbf'
+  },
+  mark: {
+    backgroundColor: '#bfbfbf',
+    height: 8,
+    width: 1,
+    marginTop: -3
+  },
+  markActive: {
+    opacity: 1,
+    backgroundColor: 'currentColor'
+  }
+})(Slider);
+
+const NewTemplateSlider = props => {
+  const [open, setOpen] = React.useState(false);
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+  const handleClose = () => {
+    let namespace = document.getElementsByName('courseCode')[0].value;
```

Comment:
As for now, the namespace name does not correspond exactly to the course code (the namespace is `course-${course-code}`). However, I am not sure how this difference is managed in the rest of the GUI.

Reply:
It's not a problem, because the namespace is selected from a list taken from the server by a menu window.